src/js/server/apply/logic.js

```diff
@@ -45,3 +45,60 @@ exports.createApplicationFromForm = function (formData, user) {
     
   });
 };
+
+exports.createTeamFromForm = function (formData, user) {
+  const members = new Set();
+  const hackerIds = [];
+  // Ensure application slugs are unique and not the applicant's own
+  return user.getHackerApplication().then(application => {
+    // Start off with the current hacker's application slug — we already know they're in the team
+    members.add(application.applicationSlug);
+  }).then(new Promise((resolve, reject) => {
+    for (const applicationSlug of [formData.memberB, formData.memberC, formData.memberD].map(s => s.trim()).filter(s => s !== '')) {
+      if (!members.has(applicationSlug)) {
+        members.add(applicationSlug);
+      } else {
+        throw new Error('Application slugs must be distinct.');
+      }
+    }
+    resolve();
+  })).then(() => {
+    const applicationSlugs = Array.from(members);
+    if (applicationSlugs.length > 1) {
+      return Promise.all(applicationSlugs.map(applicationSlug => {
+        return HackerApplication.findOne({
+          where: { applicationSlug }
+        }).then(application => {
+          if (application === null) {
+            // The application slug was not valid
+            throw new Error('The application slug matched no hacker.');
+          }
+          return application;
+        }).then(application => {
+          hackerIds.push(application.hackerId);
+          return TeamMember.findOne({
+            where: { hackerId: application.hackerId }
+          }).then(application => {
+            if (application !== null) {
+              // The hacker is already part of another team
+              throw new Error('A team member can\'t belong to more than one team.');
+            }
+          });
+        });
+      }));
+    } else {
+      throw new Error('You need at least two team members to form a team.');
+    }
+  }).then(() => {
+    // Create a new team
+    Team.create({ }).then(team => {
+      // Add the team members to the team
+      for (const hackerId of hackerIds) {
```

Comment:
Could we do this in one swoop a la

http://docs.sequelizejs.com/en/latest/docs/associations/#creating-elements-of-a-hasmany-or-belongstomany-association

This has a few benefits:

- Makes it easy to return the result in the promise so we can feed back to the user if something goes wrong
- (correct me if I'm wrong) if something goes wrong at the DB layer, the team won't be created at all (as opposed to us getting half a team)